src/Guide/Views.hs

```diff
@@ -913,3 +948,35 @@ findLinksItem item = findLinksMD item' ++ maybeToList (item^.link)
 -- | Find all Markdown links in /any/ structure, using generics.
 findLinksMD :: Data a => a -> [Url]
 findLinksMD a = [url | MD.LINK url _ <- universeBi a]
+
+data ArchiveOrgResponse =
+  ArchiveOrgResponse { archivedSnapshots :: ArchivedSnapshot
+                     } deriving (Show, Generic)
+
+data ArchivedSnapshot =
```

Comment:
I think that you don't really need all those datatypes here, because the archive.org API is pretty poor and we only need to get one thing out of it:

``` haskell
data ArchivedSnapshot {
  snapAvailable :: Bool,
  snapUrl       :: Url,
  snapTimestamp :: UTCTime,
  snapStatus    :: Text }

instance FromJSON ArchivedSnapshot where
  parseJSON = withObject "ArchivedSnapshot" $ \o -> do
    snapAvailable <- o .: "available"
    snapUrl       <- o .: "url"
    snapTimestamp <- o .: "timestamp" >>=
                     parseTimeM True defaultTimeLocale "%Y%m%d%H%M%S"
    snapStatus    <- o .: "status"
    pure ArchivedSnapshot{..}
```

Btw, with `parseTimeM` we won't `error` out if we can't parse the timestamp, which is nice.

Then you need to parse the JSON with a manual parser, which is quite easy once you have a utility function for it (which for whatever reason isn't included into `aeson` proper). I'll go and add one to `Guide.Utils` myself soon and then update this PR.